src/commonTest/kotlin/uk/gov/hmrc/CalculatorTest.kt

```diff
@@ -71,4 +76,103 @@ class CalculatorTest {
             Calculator(regularPayment = 51).run()
         }
     }
+
+    @Test
+    fun `return month breakdown if they have only saved in the first 2 months`() {
+        val calculator = Calculator(
+            regularPayment = 50,
+            currentBalance = 100,
+            currentFirstPeriodBonus = 50.0,
+            currentSecondPeriodBonus = 0.0,
+            accountStartDate = DateTime.now().minus(
+                MonthSpan(2)))
+        assertEquals(
+            2400, calculator.run().finalBalance)
+
+        assertEquals(
+            600.0, calculator.run().finalSecondYearBonus)
+        assertEquals(
+            600.0, calculator.run().finalFourthYearBonus)
+        assertEquals(
+            1200.0, calculator.run().finalTotalBonus)
+    }
+
+    @Test
+    fun `return month breakdown if they have not saved in the first 2 months`() {
+        val calculator = Calculator(
+            regularPayment = 50,
+            currentBalance = 0,
+            currentFirstPeriodBonus = 0.0,
+            currentSecondPeriodBonus = 0.0,
+            accountStartDate = DateTime.now().minus(
+                MonthSpan(2)))
+        assertEquals(
+            2300, calculator.run().finalBalance)
+
+        assertEquals(
+            550.0, calculator.run().finalSecondYearBonus)
+        assertEquals(
+            600.0, calculator.run().finalFourthYearBonus)
+        assertEquals(
+            1150.0, calculator.run().finalTotalBonus)
+    }
+
+    @Test
+    fun `return month breakdown if they have not saved in the first 24 months`() {
+        val calculator = Calculator(
+            regularPayment = 50,
+            currentBalance = 0,
+            currentFirstPeriodBonus = 0.0,
+            currentSecondPeriodBonus = 0.0,
+            accountStartDate = DateTime.now().minus(
+                MonthSpan(24)))
+        assertEquals(1200, calculator.run().finalBalance)
+
+        assertEquals(0.0, calculator.run().finalSecondYearBonus)
+        assertEquals(600.0, calculator.run().finalFourthYearBonus)
+        assertEquals(600.0, calculator.run().finalTotalBonus)
+    }
+
+    @Test
+    fun `Throw Exception when currentBalance supplied but no accounts start date`() {
+        assertFailsWith<InvalidAccountStartDate> {
+            Calculator(regularPayment = 50, currentBalance = 1000).run()
+        }
+    }
+
+    @Test
+    fun `Throw Exception when currentBalance supplied & accounts start date but no current balance amount for either period`() {
+        assertFailsWith<InvalidCurrentBonusAmount> {
+            Calculator(regularPayment = 50, currentBalance = 1000, accountStartDate = DateTime.now()).run()
+        }
+    }
+
+    @Test
+    fun `Throw Exception when currentBalance supplied & accounts start date but no first balance amount`() {
```

Comment:
Are these tests named correctly? Should they be
```suggestion
    fun `Throw Exception when currentBalance supplied & account start date but no first bonus amount`() {
```
according to the error thrown and the missing parameter? Or have I misunderstood?